### DISCUSSION TITLE
Workfllow Backup

### DISCUSSION THREAD

**Discussion by komalphadatare** [StackStorm/st2#5649] [Q&A]
Hi,

 How to take backup of StackStorm workflows and Restore ?

Regards,
Komal P

**Comment by mickmcgrath13** [comment]
Hi @komalphadatare, 

The best way by far to ensure packs are backed up is to keep them up-to-date with source control by default.  This means that changes to packs should go through source control (or if you're making modifications in the UI, copy the workflow config from the UI and update source control) and then re-install from the repo.
This way, they're essentially always backed up, and restoring would be a simple pack install for each needed pack.

An alternative would be to copy the pack contents (i.e. metadata, scripts, etc) from `/opt/stackstorm/packs/${YOUR_PACK}` and push them to an external location.  UI created workflows have a `.meta.yaml` file in the pack's `actions` directory (i.e. `/opt/stackstorm/packs/${YOUR_PACK}/actions`), and the actual workflow contents are in `/opt/stackstorm/packs/${YOUR_PACK}/actions/workflows`.

Hope this helps.
